### PR TITLE
Add draft cookie banner

### PR DESCRIPTION
**Pull request by igloosi**
This PR: 
Adds a cookie banner component styles
Adds cookie banner javascript
Adds new helper mixins
Adds spacing variables
Add gulp javascript build tasks
Modifies preview template

**Screenshot**
screen shot 2017-06-01 at 11 18 49

### CODE REVIEW

File: src/globals/scss/_vars.scss [hunk 1]
```diff
@@ -1 +1,3 @@
 $global-images: "/images/";
+$site-width: 960px;
```
Comment (gemmaleigh) [inline src/globals/scss/_vars.scss]: I think we should use the `govuk` variable prefix here, can we add it for $global-images too as it is missing.

File: src/globals/scss/_spacing.scss [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Measurements used across GOV.UK
+$one-quarter: $full-width / 4;
+$one-third: $full-width / 3;
+$half: $full-width / 2;
+$two-thirds: ($full-width) - ($one-third);
+$three-quarters: ($full-width) - ($one-quarter);
+
+$gutter: 30px;
```
Comment (gemmaleigh) [inline src/globals/scss/_spacing.scss]: Would it be possible to only copy over the vars that are used?
That way we can get rid of the ones we don't need. 

These will need a `govuk-` prefix too, can we keep them in `vars` for now and we split them out later.

File: src/components/cookie-banner/_cookie-banner.scss [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  .govuk-c-cookie-banner__text {
+    @include site-width-container;
+    @include core-16;
+    margin-top: 0;
+    margin-bottom: 0;
+  }
+
+  @include mq($media-type: print) {
```
Comment (gemmaleigh) [inline src/components/cookie-banner/_cookie-banner.scss]: Nice! :)

Comment (gemmaleigh) [inline src/components/cookie-banner/_cookie-banner.scss]: I've added the missing docs for this in a new PR.

Comment (igloosi) [inline src/components/cookie-banner/_cookie-banner.scss]: yeah saw, thanks 💯

File: src/globals/scss/_spacing.scss (hunk 2 above)
Comment (igloosi) [inline src/globals/scss/_spacing.scss]: will amend

**Comment by gemmaleigh** [comment]
👍

**Comment by gemmaleigh** [comment]
One tiny thing! Can you please rebase to amend the commits with conflicts and then delete the commits fixing the merge.

**Comment by igloosi** [comment]
closing this, opened #30

**Pull request closed without merging**